worker/rhpv3.go

```diff
@@ -424,17 +610,18 @@ func (w *worker) preparePriceTableContractPayment(hk types.PublicKey, revision *
 	}
 }
 
-// preparePriceTableAccountPayment prepare a payment function to pay for a
-// price table from the given host using the provided revision.
-// This is the preferred way of paying for a price table since it is faster and
-// doesn't require locking a contract.
-func (w *worker) preparePriceTableAccountPayment(hk types.PublicKey) PriceTablePaymentFunc {
+// preparePriceTableAccountPayment prepare a payment function to pay for a price
+// table from the given host using the provided revision.
+//
+// NOTE: This is the preferred way of paying for a price table since it is
+// faster and doesn't require locking a contract.
+func (w *worker) preparePriceTableAccountPayment(hk types.PublicKey, bh uint64) PriceTablePaymentFunc {
 	return func(pt rhpv3.HostPriceTable) (rhpv3.PaymentMethod, error) {
 		// TODO: gouging check on price table
 
 		accountKey := w.accounts.deriveAccountKey(hk)
 		account := rhpv3.Account(accountKey.PublicKey())
-		payment := rhpv3.PayByEphemeralAccount(account, pt.UpdatePriceTableCost, math.MaxUint64, accountKey)
+		payment := rhpv3.PayByEphemeralAccount(account, pt.UpdatePriceTableCost, bh+defaultWithdrawalExpiryBlocks, accountKey)
```

Comment:
I think we can reuse `preparePayment` here and move the `TODO` into `preparePayment` as well.

Reply:
I don't like what we are doing with the payment stuff currently. We have 
- `preparePayment`
- `preparePriceTableAccountPayment`
- `preparePriceTableContractPayment`
- flows doing it manually

I'm going to suggest an alternative in a separate PR if that's ok with you. Not a big PR, just making sure everyone uses the same methods and I'll add the gouging checks while I'm at it.

Reply:
can't use `preparePayment` here because I don't have access to a `worker`, probably fine to leave as is for now